Packs/TrustwaveFusion/Integrations/TrustwaveFusion/TrustwaveFusion.py

```diff
@@ -0,0 +1,587 @@
+        return int(arg)
+    if isinstance(arg, str):
+        # we use dateparser to handle strings either in ISO8601 format, or
+        # relative time stamps.
+        # For example: format 2019-10-23T00:00:00 or "3 days", etc
+        date = dateparser.parse(arg, settings={"TIMEZONE": "UTC"})
+        if date is None:
+            # if d is None it means dateparser failed to parse it
+            raise ValueError(f"Invalid date: {arg_name}")
+
+        return int(date.replace(tzinfo=timezone.utc).timestamp())
+    if isinstance(arg, (int, float)):
+        # Convert to int if the input is a float
+        return int(arg)
+    raise ValueError(f'Invalid date: "{arg_name}"')
+
+
+def format_notes(notes, limit=5):
+    """Format notes json from the Fusion API"""
+    results = []
+    for note in notes:
+        actor = note.get("actor", "")
+        ts = note.get("timestamp", "")
+        text = note.get("text", "")
+        results.append(f"{ts} Created by: {actor}")
+        results.append("NOTE:")
+        for line in text.split("\n"):
+            results.append(f"{line}")
+        results.append("----------------")
+    if results:
+        results.pop()
+    return "\n".join(results)
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client) -> str:
+    """Tests API connectivity and authentication'
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+
+    :type client: ``Client``
+    :param Client: client to use
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+
+    message: str = ""
+    try:
+        client.describe()
+        message = "ok"
+    except DemistoException as e:
+        if e.res is not None and e.res.status_code == 401:
+            message = "Authorization Error: make sure API Key is correctly set"
+        elif "Forbidden" in str(e) or "Authorization" in str(e):
+            message = "Authorization Error: make sure API Key is correctly set"
+        else:
+            raise e
+    return message
+
+
+def fetch_incidents(client, max_results, first_fetch):
+    demisto.debug(f"Fetching {max_results}, {first_fetch} {type(first_fetch)}")
+    last_run = demisto.getLastRun()
+    last_fetch = last_run.get("last_fetch", None)
+    demisto.debug(f"last_run: {last_run}")
+    if last_fetch is None:
+        # if missing, use what provided via first_fetch
+        last_fetch = first_fetch
+    else:
+        # otherwise use the stored last fetch
+        last_fetch = int(last_fetch)
+
+    search_since = max(first_fetch, last_fetch)
+    demisto.debug(f"{first_fetch} <=> {last_fetch} == {search_since}")
+    created_since = timestamp_to_datestring(search_since * 1000, FUSION_DATE_FORMAT)
+
+    ticket_types = demisto.params().get("ticket_types")
+    incidents = []
+    try:
```

Comment:
no need to wrap it in try/except (to add try/except in the function), its already wrapped in the main function, and its preferable for exceptions to be caught in the main function which you already handled there.
applies for all other functions.